service/controller/azureconfig/handler/vnetpeering/create.go

```diff
@@ -0,0 +1,225 @@
+package vnetpeering
```

Comment:
Can this be implemented under `azurecluster` controller? since `azureconfig` should go away soon

Reply:
I guess it's possible, yeah. 
I just backported the old PR to the current master to avoid being too much behind when we'll finally merge this.
Will request a review and rework what's needed when we'll be ready with the SP update.